File: 2021/day01/day01.mjs
import { promises } from "fs";

const input = (await promises.readFile("./day01.input.txt", { encoding: "utf8" }))
  .split("\n")
  .map((measurment) => parseInt(measurment))
  .slice(0, -1)

const increases = (nums) =>
  nums
    .slice(1)
    .map((curr, i) => (curr > nums[i] ? 1 : 0))
    .reduce((x, y) => x + y, 0);

const sum3 = (nums) =>
  nums
    .slice(2)
    .reduce((acc, _, i) => [...acc, nums.slice(i, i + 3)], [])
    .map((chunk) => chunk.reduce((acc, num) => acc + num, 0))

console.log(`Part 1 answer is: ${increases(input)}`);
console.log(`Part 2 answer is: ${increases(sum3(input))}`);

File: 2021/day01/day01.py
input_arr = open("day01.input.txt", "r").read().splitlines()
measurements = list(map(int, input_arr))


def count_increases(nums):
    increases = 0
    for i, v in enumerate(nums[1:]):
        if v > nums[i]:
            increases += 1
    return increases


def add_3(nums):
    acc = []
    for i in range(len(nums[2:])):
        acc.append(sum(nums[i:i + 3]))
    return acc


print('Part 1 answer is: {}'.format(count_increases(measurements)))
print('Part 2 answer is: {}'.format(count_increases(add_3(measurements))))
